# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/vsftpd/findlibs.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

diff --git a/vsf_findlibs.sh b/vsf_findlibs.sh
index f5d485d..baf167b 100755
--- a/vsf_findlibs.sh
+++ b/vsf_findlibs.sh
@@ -6,8 +6,6 @@ find_func() { egrep $1 $2 >/dev/null; }
 
 if find_func hosts_access tcpwrap.o; then
   echo "-lwrap";
-  locate_library /lib/libnsl.so && echo "-lnsl";
-  locate_library /lib64/libnsl.so && echo "-lnsl";
 fi
 
 # Look for PAM (done weirdly due to distribution bugs (e.g. Debian) or the
@@ -36,7 +34,6 @@ locate_library /lib/libdl.so && echo "-ldl";
 locate_library /lib/libsocket.so && echo "-lsocket";
 
 # Look for libnsl. Solaris needs this.
-locate_library /lib/libnsl.so && echo "-lnsl";
 
 # Look for libresolv. Solaris needs this.
 locate_library /lib/libresolv.so && echo "-lresolv";
@@ -69,7 +66,7 @@ locate_library /usr/shlib/librt.so && echo "-lrt";
 locate_library /usr/lib/libsendfile.so && echo "-lsendfile";
 
 # OpenSSL
-if find_func SSL_library_init ssl.o; then
+if find_func SSL_CTX_new ssl.o; then
   echo "-lssl -lcrypto";
 fi
 
